### PR TITLE
RELATED: ONE-3393 Fixed invalid color assignment

### PR DESCRIPTION
**Pull request by ZdenekVidensky**

### CODE REVIEW

**Comment by ZdenekVidensky** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2221/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1656/ : SUCCESS in 1m 37s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/897/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2224/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/511/ : UNSTABLE in 17m 38s

**Comment by ZdenekVidensky** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2225/ : SUCCESS in 56s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1658/ : SUCCESS in 3m 48s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/898/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2226/ : SUCCESS in 58s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/512/ : SUCCESS in 19m 11s

File: src/components/visualizations/chart/colorFactory.ts [hunk 1]
```diff
@@ -260,7 +262,8 @@ function getAtributeColorAssignment(
     return uniqItems.map((headerItem) => {
         const mappedColor = getColorFromMapping(headerItem, colorMapping);
 
-        const color: IColorItem = mappedColor ? mappedColor
+        const color: IColorItem = mappedColor && isValidMappedColor(mappedColor, colorPalette)
```
Comment (pbenes) [inline src/components/visualizations/chart/colorFactory.ts]: just thinking, if we would have 
`if(!mappedColor) { return false; }` at the very beginining of `isValidMappedColor` method, we could avoid having `mappedColor &&` here and also on other places (the same here https://github.com/gooddata/gooddata-react-components/pull/674/files#diff-934a24e28244a84cb0a91e27b0b8aee2R442)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2228/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1660/ : SUCCESS in 1m 45s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/490/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1661/ : SUCCESS in 1m 42s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/899/ : SUCCESS in 14m 48s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/434/ : SUCCESS in 22m 14s

**Pull request merged**